generate/src/Spec/StripExtensions.hs

```diff
@@ -28,10 +31,16 @@ stripWSIExtensions spec =
       isInDisallowed  = flip elem disallowedTypes
       allowedTypeDecls = filter (not . isInDisallowed) allTypes
       isInDisallowedNames = flip elem disallowedTypeNames
-      allowedCommands = filter
-                         (not . any isInDisallowedNames . commandDependencies)
-                         allCommands
-      allowedExtensions = []
+      (disallowedCommands, allowedCommands) = partition
+                                                (any isInDisallowedNames . commandDependencies)
+                                                allCommands
+      disallowedCommandNames = fmap cName disallowedCommands
+      isInDisallowedCommands = flip elem disallowedCommandNames
+      allowedExtensions = filter
+                            (pure (&&)
```

Comment:
Err yeah, I'll fix that.
> @@ -28,10 +31,16 @@ stripWSIExtensions spec =
>        isInDisallowed  = flip elem disallowedTypes
>        allowedTypeDecls = filter (not . isInDisallowed) allTypes
>        isInDisallowedNames = flip elem disallowedTypeNames
> -      allowedCommands = filter
> -                         (not . any isInDisallowedNames . commandDependencies)
> -                         allCommands
> -      allowedExtensions = []
> +      (disallowedCommands, allowedCommands) = partition
> +                                                (any isInDisallowedNames . commandDependencies)
> +                                                allCommands
> +      disallowedCommandNames = fmap cName disallowedCommands
> +      isInDisallowedCommands = flip elem disallowedCommandNames
> +      allowedExtensions = filter
> +                            (pure (&&)

Would it be neater to use `(&&) <$> not...eTypeNames <*> not...eCommandNames`?

---
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub:
https://github.com/expipiplus1/vulkan/pull/8/files/a558e534d2546f1075651b1c2d68520b65d3bccf#r57877332